Sources/Sentry/SentryPerformanceTracker.m

```diff
@@ -0,0 +1,182 @@
+#import "SentryPerformanceTracker.h"
+#import "SentryLog.h"
+#import "SentrySDK+Private.h"
+#import "SentrySpan.h"
+#import "SentrySpanProtocol.h"
+#import "SentryTracer.h"
+
+static NSString *const SENTRY_PERFORMANCE_TRACKER_SPANS = @"SENTRY_PERFORMANCE_TRACKER_SPANS";
+static NSString *const SENTRY_PERFORMANCE_TRACKER_ACTIVE_STACK
+    = @"SENTRY_PERFORMANCE_TRACKER_ACTIVE_STACK";
+
+/*
+ * Auxiliary class to store tracking information.
+ */
+@interface SentrySpanTracker : NSObject
+
+@property (nonatomic) id<SentrySpan> span;
+@property (nonatomic) BOOL finished;
+@property (nonatomic) SentrySpanStatus finishedStatus;
+@property (nonatomic, strong) NSMutableDictionary<NSString *, SentrySpanTracker *> *children;
+
+- (instancetype)initWithSpan:(id<SentrySpan>)span;
+- (void)markFinishedWithStatus:(SentrySpanStatus)status;
+
+@end
+
+@implementation SentrySpanTracker
+
+- (instancetype)initWithSpan:(id<SentrySpan>)span
+{
+    if (self = [super init]) {
+        self.span = span;
+        self.finished = false;
+        self.children = [[NSMutableDictionary alloc] init];
+    }
+    return self;
+}
+- (void)markFinishedWithStatus:(SentrySpanStatus)status
+{
+    self.finishedStatus = status;
+    self.finished = YES;
+}
+@end
+
+@implementation SentryPerformanceTracker
+
++ (instancetype)shared
+{
+    static SentryPerformanceTracker *instance = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{ instance = [[self alloc] init]; });
+    return instance;
+}
+
+/**
+ * A dictionary of created spans in this thread using id as Key.
+ */
+- (NSMutableDictionary *)spansForThread
+{
+    NSMutableDictionary *res =
```

Comment:
`l`: If you like to use the pattern to name the variable `result` I would prefer naming actually `result` instead of `res`. I was a bit confused by that.
```suggestion
    NSMutableDictionary *result =
```